immuni_common/helpers/logging.py

```diff
@@ -12,37 +12,143 @@
 #  along with this program. If not, see <https://www.gnu.org/licenses/>.
 
 import logging
+import re
 import sys
+from typing import Any, Dict, Optional, Set
 
+from gunicorn import glogging
 from pythonjsonlogger import jsonlogger
+from pythonjsonlogger.jsonlogger import RESERVED_ATTRS
+from sanic.log import LOGGING_CONFIG_DEFAULTS
 
 from immuni_common.models.enums import LogLevel
 
 
-def initialize_logging(log_level: LogLevel) -> None:
+class RedactingJsonFormatter(jsonlogger.JsonFormatter):
```

Comment:
can't you just define the logging dictionary for both the Gunicorn and Sanic logging configuration, instead of implementing these classes?  https://docs.python.org/3/library/logging.config.html#configuration-dictionary-schema 
It should be easier and more readable, as it should be enough to define the custom Python formatters and assign them to the handlers.
Also, on Sanic you can just disable the access_log when running in the production environments by passing access_log=False to the `app.run` method.

Reply:
> can't you just define the logging dictionary for both the Gunicorn and Sanic logging configuration, instead of implementing these classes?

**Gunicorn**
In that case, you'd need to make sure the `logging.config.dictConfig` method is called before gunicorn starts. Personally, I found the `--logger-class` command line argument to be passed to  gunicorn more explicit, thus easier to read. It's a personal preference, so we could converge based on the other comments.

**Sanic**
Specifying the log config while initializing the Sanic app seems to be the recommended way in their documentation. Beneath, it eventually calls `logging.config.dictConfig`.
The current implementation aims at preserving any change the Sanic library would do to their loggers, through `sanic.log.LOGGING_CONFIG_DEFAULTS`, only overriding their formatters.
I'm not sure how exactly you envision the single configuration.

> it should be enough to define the custom Python formatters and assign them to the handlers

I was afraid this would not cover any additional logger/handler Sanic would add/remove on its side.

> Also, on Sanic you can just disable the access_log when running in the production environments by passing access_log=False to the app.run method.

IMHO, this is not really needed as in production we run through gunicorn, which already takes care of inhibiting Sanic's original loggers.
The `app.run` in `run_app` should be only used in development.

Reply:
I would explain why we had to create this class instead of leveraging logging dicts. It is to address an issue with the current Uvicorn implementation, which does not offer a proper way of overriding the formatter configuration.

Reply:
We agreed upon simplifying the implementation for Sanic. Although the original implementation allowed for flexibility in case of Sanic formatters changes, and given the odds of the latter being relatively low, we opted for a more explicit and readable configuration.

As always, if not asked otherwise, KISS. 😘